File: src/main/java/io/izzel/taboolib/module/db/sql/query/JoinType.java
package io.izzel.taboolib.module.db.sql.query;

/**
 * io.izzel.taboolib.module.db.sql.query.JoinType
 *
 * @author sky
 * @since 2021/4/16 9:08 上午
 */
public enum JoinType {

    LEFT, RIGHT, INNER;

    public String toKey() {
        return name().toLowerCase();
    }
}

File: module/module-database/src/main/kotlin/taboolib/module/database/JoinType.kt
package taboolib.module.database

/**
 * 连接类型
 *
 * @author sky
 * @since 2021/6/23 3:15 下午
 */
enum class JoinType {

    /**
     * 左连接（左表的每一行都会加上右表中符合条件的行）
     */
    LEFT,

    /**
     * 右连接（右表的每一行都会加上左表中符合条件的行）
     */
    RIGHT,

    /**
     * 内连接（两表的交集）
     */
    INNER
}